### PR TITLE
Reuse validation enqueuer

### PR DESCRIPTION
**Pull request by Dref360**
Fixes https://github.com/keras-team/keras/issues/10474

Attempt at reusing the validation enqueuer (I thought we were already doing that).

Not sure how to test this. Suggestion welcome.

### CODE REVIEW

**Comment by lexxxel** [comment]
This commit does not fix the problem, yet.

**Comment by Dref360** [comment]
It's WIP

**Comment by lexxxel** [comment]
With this patch on the second epoch the validation gets slow again, but the strange thing is that there is a huge 1cpu workload after the validation finished that blocks the start of the next epoch.
Also saving of the model is broken. 
```
 File "/home/apit/.local/lib/python3.6/site-packages/keras/engine/network.py", line 1104, in save
    save_model(self, filepath, overwrite, include_optimizer)
  File "/home/apit/.local/lib/python3.6/site-packages/keras/engine/saving.py", line 106, in save_model
    f = h5py.File(filepath, mode='w')
  File "/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py", line 312, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py", line 148, in make_fid
    fid = h5f.create(name, h5f.ACC_TRUNC, fapl=fapl, fcpl=fcpl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 98, in h5py.h5f.create
OSError: Unable to create file (unable to lock file, errno = 11, error message = 'Resource temporarily unavailable')
```

**Comment by lexxxel** [comment]
Sry, I appreciate your help and just tried to help you with debug information.

**Comment by Dref360** [comment]
As for the single core being used, it's probably that the queue is full so the other workers are not doing anything.

The model saving issue is not related to this PR.

**Comment by Dref360** [comment]
I made some tests and seems to gets some valuable improvements when using large objects.

| | Before | After |
|-| ------ | ------|
|Generator| 67.87834 | 47.59921 |
| Sequence| 70.4015 | 48.8415|

I also validate that all cores are used in this example. The enqueuer is also closed at the end. 

## Repro

```python
import time

import numpy as np

from keras.layers import Dense
from keras.models import Sequential
from keras.utils import Sequence


class SequenceGen(Sequence):
    def __init__(self, batch_size):
        self.super_large = ['rick'] * 100000
        self.batch_size = batch_size

    def __len__(self):
        return 20  # dummy

    def __getitem__(self, idx):
        batch_x = np.zeros((self.batch_size, 100))
        batch_y = np.zeros((self.batch_size, 1))
        time.sleep(0.1)
        return batch_x, batch_y


def super_gen(seq):
    # Creates a gen from a Sequence
    while True:
        for i in range(len(seq)):
            yield seq[i]


data_generator = SequenceGen(batch_size=20)
val_generator = SequenceGen(batch_size=20)


# Dummy Model
model = Sequential()

model.add(Dense(100, input_shape=(100,)))
model.add(Dense(units=25))
model.add(Dense(units=1))

model.compile(optimizer='sgd', loss='mae')

# Timing for Sequence
start = time.time()
model.fit_generator(
    generator=data_generator,
    epochs=100,
    verbose=0,
    validation_data=val_generator,
    max_queue_size=10,
    workers=5,
    use_multiprocessing=True)
end = time.time()
print("SEQUENCE", end - start)

# Timing for Generator
start = time.time()
model.fit_generator(
    generator=super_gen(data_generator),
    steps_per_epoch=len(data_generator),
    epochs=100,
    verbose=0,
    validation_data=super_gen(val_generator),
    validation_steps=len(val_generator),
    max_queue_size=10,
    workers=5,
    use_multiprocessing=True)
end = time.time()
print("Generator", end - start)

```

File: tests/keras/test_callbacks.py [hunk 1]
```diff
@@ -547,7 +547,7 @@ def callbacks_factory(histogram_freq, embeddings_freq=1):
 @keras_test
 @pytest.mark.skipif((K.backend() != 'tensorflow'),
                     reason='Requires TensorFlow backend')
-def test_TensorBoard_histogram_freq_must_have_validation_data(tmpdir):
+def pipq(tmpdir):
```
Comment (fchollet) [inline tests/keras/test_callbacks.py]: Why rename this test? If it isn't prefixed by "test", it will not be run by pytest.

Comment (Dref360) [reply tests/keras/test_callbacks.py]: Typo, fixed

**Pull request merged**